meilisearch-http/src/backup.rs

```diff
@@ -0,0 +1,420 @@
+        if values.len() == document_batch_size { 
+            let batch = std::mem::replace(&mut values, Vec::with_capacity(document_batch_size));
+            apply_documents_addition(write_txn, &index, batch)?;
+        }
+    }
+
+    // apply documents remaining in the buffer 
+    if !values.is_empty() { 
+        apply_documents_addition(write_txn, &index, values)?;
+    }
+
+    Ok(())
+}
+
+/// Import backup from `backup_path` in database.
+pub fn import_backup(
+    data: &Data,
+    backup_path: &Path,
+    document_batch_size: usize,
+) -> Result<(), Error> {
+    // create a temporary directory
+    let tmp_dir = TempDir::new()?;
+    let tmp_dir_path = tmp_dir.path();
+
+    // extract backup in temporary directory
+    compression::from_tar_gz(backup_path, tmp_dir_path)?;
+
+    // read backup metadata
+    let metadata = BackupMetadata::from_path(&tmp_dir_path)?;
+
+    // choose importation function from BackupVersion of metadata
+    let import_index = match metadata.backup_version {
+        BackupVersion::V1 => import_index_v1,
+    };
+
+    // remove indexes which have same `uid` than indexes to import and create empty indexes
+    let existing_index_uids = data.db.indexes_uids();
+    for index in metadata.indexes.iter() {
+        if existing_index_uids.contains(&index.uid) {
+            data.db.delete_index(index.uid.clone())?;
+        }
+        index::create_index_sync(&data.db, index.uid.clone(), index.name.clone(), index.primary_key.clone())?;
+    }
+
+    // import each indexes content
+    data.db.main_write::<_, _, Error>(|mut writer| {
+        for index in metadata.indexes {
+            import_index(&data, tmp_dir_path, &index.uid, document_batch_size, &mut writer)?;
+        }
+        Ok(())
+    })?;
+
+    Ok(())
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum BackupStatus {
+    Done,
+    Processing,
+    BackupProcessFailed,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct BackupInfo {
+    pub uid: String,
+    pub status: BackupStatus,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub error: Option<String>,
+}
+
+impl BackupInfo {
+    pub fn new(uid: String, status: BackupStatus) -> Self {
+        Self { uid, status, error: None }
+    }
+
+    pub fn with_error(mut self, error: String) -> Self {
+        self.status = BackupStatus::BackupProcessFailed;
+        self.error = Some(error);
+
+        self
+    }
+
+    pub fn backup_already_in_progress(&self) -> bool {
+        self.status == BackupStatus::Processing
+    }
+
+    pub fn get_current() -> Option<Self> {
+        BACKUP_INFO.lock().unwrap().clone()
+    }
+
+    pub fn set_current(&self) {
+        *BACKUP_INFO.lock().unwrap() = Some(self.clone());
+    }
+}
+
+/// Generate uid from creation date
+fn generate_uid() -> String {
+    Utc::now().format("%Y%m%d-%H%M%S").to_string()
```

Comment:
You kept a `-`, is it ok with @curquiza?

Reply:
I think `-` is acceptable and make it easier to read, for example if I want to retrieve a backup from a specific day I don't care of the hour minute and second

Reply:
Ok for me for this kind of format with the `-` (ex: `20200923-210403`) 🙂